clippy_lints/src/option_if_let_else.rs

```diff
@@ -188,6 +195,16 @@ fn try_get_option_occurrence<'tcx>(
     None
 }
 
+/// Gets the call expr iff it does not have any args and was not from macro expansion.
+fn try_get_argless_call_expr<'tcx>(expr: &Expr<'tcx>) -> Option<&'tcx Expr<'tcx>> {
+    if !expr.span.from_expansion() &&
+        let ExprKind::Call(call_expr, []) = expr.kind
+    {
+        return Some(call_expr);
+    }
+    None
+}
```

Comment:
This snippet may help avoid a function call if `expr` is not an `ExprKind::Call` and looks more balanced IMO:

```suggestion
    match expr.kind {
        ExprKind::Call(call_expr, []) if !expr.span.from_expansion() => Some(call_expr),
        _ => None,
    }
```

Reply:
good catch! it might or might not triggers another clippy warning, I'm not sure, let's find out 😆

Reply:
there was an extra bracket at the end, but it's all good now~ thanks!